src/Perpetuum/Modules/SensorJammerModule.cs

```diff
@@ -31,14 +31,17 @@ protected override void OnAction()
             var unitLock = GetLock().ThrowIfNotType<UnitLock>(ErrorCodes.InvalidLockType);
             var robot = unitLock.Target.ThrowIfNotType<Robot>(ErrorCodes.InvalidTarget);
 
-            var targetSensorStrength = robot.SensorStrength * FastRandom.NextDouble();
-            targetSensorStrength = ModifyValueByOptimalRange(robot, targetSensorStrength);
-
-            var success = targetSensorStrength < _ecmStrength.Value;
+            var rangeMod = ModifyValueByOptimalRange(robot, 1.0);
+            var success = FastRandom.NextDouble() <= rangeMod;
             if (success)
             {
-                robot.ResetLocks();
-                robot.AddThreat(ParentRobot, new Threat(ThreatType.Ewar, Threat.SENSOR_DAMPENER));
+                var targetSensorStrength = robot.SensorStrength * FastRandom.NextDouble() * rangeMod;
+                success = targetSensorStrength < _ecmStrength.Value;
+                if (success)
```

Comment:
I would embed these checks in the if statement rather than create and reuse the variable for two different ones

Reply:
`success` does also get reuse in the response packet.  But I'll take your point having the conditional have the boolean expression might be more clear